File: templates/fsharp/xplat/AvaloniaTest/ViewModels/ViewModelBase.fs
namespace AvaloniaTest.ViewModels

#if (CommunityToolkitChosen)
open CommunityToolkit.Mvvm.ComponentModel
#elif (ReactiveUIToolkitChosen)
open ReactiveUI
#endif

[<AbstractClass>]
type ViewModelBase() =
#if (CommunityToolkitChosen)
    inherit ObservableObject()
#elif (ReactiveUIToolkitChosen)
    inherit ReactiveObject()
#endif

File: templates/csharp/xplat/AvaloniaTest/ViewModels/ViewModelBase.cs
#if (CommunityToolkitChosen)
using CommunityToolkit.Mvvm.ComponentModel;
#elif (ReactiveUIToolkitChosen)
using ReactiveUI;
#endif

namespace AvaloniaTest.ViewModels;

#if (CommunityToolkitChosen)
public abstract class ViewModelBase : ObservableObject
#elif (ReactiveUIToolkitChosen)
public abstract class ViewModelBase : ReactiveObject
#endif
{
}
